packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    CarouselHandledProps,
+    CarouselProps,
+    CarouselUnhandledProps,
+    Slide,
+    SlideTheme,
+} from "./carousel.props";
+import { Flipper, FlipperDirection } from "../flipper";
+import { Tabs } from "@microsoft/fast-components-react-base";
+import { TabsClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { get } from "lodash-es";
+
+/**
+ * The carousel state interface
+ */
+export interface CarouselState {
+    /**
+     * Holds the active tab id to share with other controls
+     */
+    activeId: string;
+}
+
+class Carousel extends Foundation<
+    CarouselHandledProps,
+    CarouselUnhandledProps,
+    CarouselState
+> {
+    public static displayName: string = "Carousel";
+
+    protected handledProps: HandledProps<CarouselHandledProps> = {
+        managedClasses: void 0,
+        label: void 0,
+        activeId: void 0,
+        items: void 0,
+    };
+
+    private slides: Slide[];
+
+    /**
+     * Define constructor
+     */
+    constructor(props: CarouselProps) {
+        super(props);
+
+        if (this.props.items) {
```

Comment:
What happens if this is just one item? Perhaps we can perform that check here instead of in the state object:

```
if (Array.isArray(this.props.items) && this.props.items.length > 0) {}
```

Reply:
Sure, that would result in this:
```
if (Array.isArray(this.props.items) && this.props.items.length > 0) {
            this.state = {
                activeId:
                    typeof this.props.activeId === "string"
                        ? this.props.activeId
                        : get(this.props.items[0], "id")
                    };
        } else {
            this.state = {
                activeId: ""
        }
```

Reply:
@chrisdholt  I actually think it was correct before, if you follow the logic again. First we want to check to make sure items exists at all. Then we want to set the id if it is passed OR get the id from the first slide OR set it to empty (because there is no first slide)
Changing the initial if parameters to check for length as well convolutes this logic and would actually cause this if/else block to be longer (even longer than I put above - it's missing another block). I think the correct solution would be this:
```
if (Array.isArray(this.props.items)) {
            this.state = {
                activeId:
                    typeof this.props.activeId === "string"
                        ? this.props.activeId
                        : this.props.items.length > 0
                            ? get(this.props.items[0], "id")
                            : "",
            };

            this.updateSlides();
        }
```
@janechu can verify the logic from `tabs`

Reply:
I think we might be able to simplify; noted below.